src/status_im/contexts/wallet/wallet_connect/effects.cljs

```diff
@@ -145,3 +145,35 @@
            :reason      reason})
          (promesa/then on-success)
          (promesa/catch on-error)))))
+
+(rf/reg-fx
+ :effects.wallet-connect/should-session-request-handled
+ (fn [{:keys [db event on-handle on-do-not-handle]}]
+   (let [{:keys [topic]} event
+         should-handle?  (some #(= topic %)
+                               (map :topic (:wallet-connect/sessions db)))]
+     (if should-handle?
+       (on-handle)
+       (do
+         (rf/dispatch [:wallet-connect/send-response
+                       {:error (wallet-connect/get-sdk-error
+                                constants/wallet-connect-user-rejected-chains-error-key)}])
+         (on-do-not-handle))))))
```

Comment:
Why is this an effect and not a simple function? We're just checking if we have a session with topic X in the db. I think there's no need to complicate it with passing callbacks.

Reply:
True, dispatching is an fx, which is why we can declare the dispatch :fx map. [Here](https://github.com/day8/re-frame/blob/master/docs/EffectfulHandlers.md#step-1-of-plan) you can see that there is a difference between explicitly calling `rf/dispatch` and declaring it in the fx map, like below:
```clj
(when should-handle?
  {:fx [[:dispatch [:wallet-connect/send-response {:error ...}]]]})
```

Also there must be a point about this somewhere in the guidelines if I remember correctly.

Reply:
Yes, but then we'll have to manually dispatch the response for each type of request. Right now there are two instances of this. There might be more in future.

With my solution, we can outsource that rejection to the to the fx.

Reply:
In this case it should be an event, not an fx

Reply:
Decision on Discord: https://discord.com/channels/1210237582470807632/1217175655729467492/1263121102268268618